File: MultiLanguageSandbox/src/test/deps/elixir/45.exs
defmodule ReverseNumber do
def reverse_three_digit(number) when is_integer(number) and number >= 100 and number <= 999 do
    hundreds = div(number, 100)
    tens = rem(div(number, 10), 10)
    units = rem(number, 10)
    
    # Reverse the digits and convert back to integer
    reversed = units * 100 + tens * 10 + hundreds
    
    # Handle cases where the reversed number has leading zeros (like 001 becomes 1)
    if reversed == 0 do
      0
    else
      reversed
    end
  end
end



ExUnit.start()
defmodule ReverseNumberTest do
  use ExUnit.Case

  test "reverse_three_digit reverses numbers correctly" do
    assert ReverseNumber.reverse_three_digit(123) == 321
    assert ReverseNumber.reverse_three_digit(100) == 1  # Note: Leading zeros are not preserved in integers
    assert ReverseNumber.reverse_three_digit(905) == 509
    assert ReverseNumber.reverse_three_digit(210) == 12  # Note: Leading zeros result in a lower integer value
  end

  test "reverse_three_digit handles edge cases" do
    assert ReverseNumber.reverse_three_digit(999) == 999
    assert ReverseNumber.reverse_three_digit(101) == 101
  end
end

File: MultiLanguageSandbox/src/test/deps/swift/45.swift
func reverseThreeDigitNumber(_ number: Int) -> Int {
    // Check if the number is a three-digit number
    if number < 100 || number > 999 {
        return -1
    }
    
    // Extract each digit
    let hundreds = number / 100
    let tens = (number / 10) % 10
    let units = number % 10
    
    // Reverse the digits
    let reversedNumber = units * 100 + tens * 10 + hundreds
    
    return reversedNumber
}


func testReverseThreeDigitNumber() {
    assert(reverseThreeDigitNumber(123) == 321, "Reverse of 123 should be 321.")
    assert(reverseThreeDigitNumber(456) == 654, "Reverse of 456 should be 654.")
    assert(reverseThreeDigitNumber(789) == 987, "Reverse of 789 should be 987.")
    assert(reverseThreeDigitNumber(100) == 1, "Reverse of 100 should be 1.")
    assert(reverseThreeDigitNumber(905) == 509, "Reverse of 905 should be 509.")
    assert(reverseThreeDigitNumber(999) == 999, "Reverse of 999 should be 999.")
    assert(reverseThreeDigitNumber(210) == 12, "Reverse of 210 should be 12.")
}

// Call the test function to verify the functionality
testReverseThreeDigitNumber()